ortools/linear_solver/linear_solver.cc

```diff
@@ -1514,19 +1514,27 @@ MPConstraint* MPSolver::MakeRowConstraint(const LinearRange& range,
 MPConstraint* MPSolver::MakeIndicatorConstraint(
     double lb, double ub, const std::string& name,
     const MPVariable* indicator_variable, bool indicator_value) {
+  if (!indicator_variable->integer() || indicator_variable->lb() != 0 ||
+      indicator_variable->ub() != 1) {
+    LOG(ERROR) << "Variable " << indicator_variable->name()
+               << " is not boolean";
+    return nullptr;
+  }
   const int constraint_index = NumConstraints();
   MPConstraint* const constraint =
       new MPConstraint(constraint_index, lb, ub, name, interface_.get());
-  // TODO: check that variable is boolean?
   constraint->indicator_variable_ = indicator_variable;
   constraint->indicator_value_ = indicator_value;
+  if (!interface_->AddIndicatorConstraint(constraint)) {
+    LOG(ERROR) << "Solver doesn't support indicator constraints";
```

Comment:
I don't think it's necessary to display a log message here, it's already done in the base version of `AddIndicatorConstraint` (linear_solver.cc)
```cpp
virtual bool AddIndicatorConstraint(MPConstraint* const /*ct*/) {
  LOG(ERROR) << "Solver doesn't support indicator constraints.";
  return false;
}
```
It would be displayed twice. Not terrible, but not great either.

Reply:
but then for interfaces that can sometimes handle indicator constraints, sometimes not (for example an interface that can handle MIP or LP problems), we'll have to log the issue before returning "false"